pkg/operator/staticpod/staticpodcontroller/installer_controller_test.go

```diff
@@ -0,0 +1,278 @@
+package staticpodcontroller
+
+import (
+	"errors"
+	"os"
+	"testing"
+	"time"
+
+	"k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/informers"
+	"k8s.io/client-go/kubernetes/fake"
+	ktesting "k8s.io/client-go/testing"
+	"k8s.io/client-go/tools/cache"
+
+	operatorv1alpha1 "github.com/openshift/api/operator/v1alpha1"
+)
+
+func TestNewNodeStateForInstallInProgress(t *testing.T) {
+	kubeClient := fake.NewSimpleClientset()
+
+	var (
+		installerPod   *v1.Pod
+		createPodCount int
+	)
+
+	kubeClient.PrependReactor("create", "pods", func(action ktesting.Action) (handled bool, ret runtime.Object, err error) {
+		installerPod = action.(ktesting.CreateAction).GetObject().(*v1.Pod)
+		createPodCount += 1
+		return false, nil, nil
+	})
+
+	kubeInformers := informers.NewSharedInformerFactoryWithOptions(kubeClient, 1*time.Minute, informers.WithNamespace("test"))
+
+	fakeStaticPodOperatorClient := &staticPodOperatorClient{
+		fakeOperatorSpec: &operatorv1alpha1.OperatorSpec{
+			ManagementState: operatorv1alpha1.Managed,
+			Version:         "3.11.1",
+		},
+		fakeOperatorStatus: &operatorv1alpha1.OperatorStatus{},
+		fakeStaticPodOperatorStatus: &operatorv1alpha1.StaticPodOperatorStatus{
+			LatestAvailableDeploymentGeneration: 1,
+			NodeStatuses: []operatorv1alpha1.NodeStatus{
+				{
+					NodeName:                    "test-node-1",
+					CurrentDeploymentGeneration: 0,
+					TargetDeploymentGeneration:  0,
+				},
+			},
+		},
+		t:               t,
+		resourceVersion: "0",
+	}
+	os.Setenv("OPERATOR_IMAGE", "docker.io/foo/bar")
```

Comment:
I make an injectable function to avoid doing this.  Looking at this test, it appears you'll need to make it a private member on the controller struct.